core/src/main/java/org/apache/calcite/avatica/remote/looker/LookerResponseParser.java

```diff
@@ -82,21 +86,25 @@ static Object deserializeValue(JsonParser parser, ColumnMetaData columnMetaData)
       return parser.getShortValue();
     case PRIMITIVE_INT:
     case INTEGER:
-      return parser.getIntValue();
+      return parser.getValueAsInt();
     case PRIMITIVE_LONG:
     case LONG:
-      return parser.getLongValue();
+      return parser.getValueAsLong();
     case PRIMITIVE_FLOAT:
     case FLOAT:
       return parser.getFloatValue();
     case PRIMITIVE_DOUBLE:
     case DOUBLE:
-      return parser.getDoubleValue();
+      return parser.getValueAsDouble();
     case NUMBER:
       // NUMBER is represented as BigDecimal
-      return parser.getDecimalValue();
-    // TODO: MEASURE types are appearing as Objects. This should have been solved by CALCITE-5549.
-    //  Be sure that the signature of a prepared query matches the metadata we see from JDBC.
+      if (parser.currentToken() == JsonToken.VALUE_STRING) {
+        return NumberInput.parseBigDecimal(parser.getValueAsString());
+      } else {
+        return parser.getDecimalValue();
+      }
+      // TODO: MEASURE types are appearing as Objects. This should have been solved by CALCITE-5549.
```

Comment:
nit: align comment with `case` below

Reply:
Also can we put a b/ or JIRA # (or are we still using 5549?)

Reply:
Will do - that comment was there previously and I didn't follow up on whether it was finished. I can investigate.